docs/load-testing.md

```diff
@@ -0,0 +1,90 @@
+# Load Testing with [`k6`](https://k6.io)
+
+**k6** is a developer-centric, free and open-source load testing tool built for making performance testing a productive and enjoyable experience. Below, we'll walk you through running a load test on Marquez using [OpenLineage](https://github.com/OpenLineage/OpenLineage) and `k6` to:
+
+* Performance test `POST` `/lineage` (HTTP requests _blocked_, _waiting_, _duration_, etc)
+* Understand query performance on database
+
+To view the built-in metrics automatically collects by `k6`, see [_Metrics_](https://k6.io/docs/using-k6/metrics/#http-specific-built-in-metrics).
+
+## Setup [`k6 `](https://k6.io)
+
+```bash
+$ brew install k6
+```
+
+## Architecture
+
+<figure align="center">
+  <img src="./load-testing-architecture.png">
+</figure>
+
+> **Figure 1:** Using [`k6`](https://k6.io) to load test Marquez with [OpenLineage](https://github.com/OpenLineage/OpenLineage) events defined in `metadata.json` generated by the [`metadata`](https://github.com/MarquezProject/marquez/blob/main/api/src/main/java/marquez/cli/MetadataCommand.java) command.
+
+#### COMPONENTS
+
+| Component        | Image                                                                               | Description                                                                                                            |
+|------------------|-------------------------------------------------------------------------------------|------------------------------------------------------------------------------------------------------------------------|
+| Marquez HTTP API | [marquezproject/marquez](https://hub.docker.com/r/marquezproject/marquez)           | The core API used to collect metadata using [OpenLineage](https://openlineage.io).                                     |
+| Database         | [bitnami/postgresql](https://hub.docker.com/r/bitnami/postgresql) or cloud provided | A PostgreSQL instance used to store metadata.                                                                          |
+
+
+## Create `metadata.json`
+
+Use the [`metadata`](https://github.com/MarquezProject/marquez/blob/main/api/src/main/java/marquez/cli/MetadataCommand.java) command to generate random _dataset_, _job_, and _run_ metadata. Below, we generate `metadata.json` with `128` runs (`20` events in total), where each `START` event will have a size of `~16384` bytes:
+
+```bash
+$ java -jar marquez-api.jar metadata --runs 10 --bytes-per-event 16384
+```
+
+> **Tip**: You may specify the location of `metadata.json` with `--output`.
+
+> **Note:** You may need to manually [build](https://github.com/MarquezProject/marquez/tree/feature/load-testing-docs#building) `marquez-api.jar`.
+
+## Create `load.js`
+
+```js
+import { SharedArray } from 'k6/data';
```

Comment:
Consider placing this code in Marquez repo.

Reply:
Yep! I'll add it as part of CI